src/mkdocstrings/plugin.py

```diff
@@ -37,6 +37,8 @@
 """The name of the selection parameter in YAML configuration blocks."""
 RENDERING_OPTS_KEY: str = "rendering"
 """The name of the rendering parameter in YAML configuration blocks."""
+DISABLE_PLUGIN_ENV_VAR: str = "DISABLE_MKDOCSTRINGS" 
+"""The name of the environment variable that disables mkdocstrings."""
```

Comment:
Rather than checking the environment in each hook, could we instead check it once here and use the value in the hooks?

```python
DISABLED: str =  os.getenv("DISABLE_MKDOCSTRINGS", "0").lower() in ("y", "yes", "true", "on", "1")
```

Then in the hooks:

```python
if DISABLED:
    return
```

I can't find a scenario where the environment variable would be updated in-between hooks. That would probably break everything anyway.

If assigning the value at import time is too early, we could maybe do it in the plugin init method instead.

WDYT?

Reply:
I think that I don't know enough about the inner workings of mkdocs to say for sure how early it can be checked, so I just set it to be done as late as possible to be sure.

We could also log the debug message in the `on_startup` hook instead.

Reply:
That's irrelevant now that we want to add the config option anyway, resolving :)

Reply:
There's still the issue of multiple debug statements. I guess we could either:

- Don't log anything
- Log in every hook and don't care that we log multiple times
- Somehow check if we have logged the message already

I think the third one might be difficult, because when using `mkdocs serve` the user can change back and forth.

I think it's nice to get the debug message, just so it's very clear that the plugin is deliberately not being run.

But I have no strong feelings either way, I'll just make the PR without them and then they can be added in if needed later :)

Reply:
Why not logging in the earliest hook and stop there?

Reply:
Oh yeah, I was thinking about only logging once per session. Logging in `on_config` should be good enough, right?

Reply:
Yes, that's completely fine to have the debug log appear at each reload :slightly_smiling_face: